### PR TITLE
Fix Queue capability overuse when specify minAvailable  less than task replicas

### PR DESCRIPTION
**Pull request by zen-xu**
Ref issue #834.
I have checked my pull request can fix this bug.

### CODE REVIEW

**Comment by hzxuzhonghu** [comment]
@zen-xu Isn't this fixed by #959

**Comment by zen-xu** [comment]
Yes

**Comment by zen-xu** [comment]
> @zen-xu Isn't this fixed by #959
Yes

**Comment by hzxuzhonghu** [comment]
So i am confused what's this for

**Comment by zen-xu** [comment]
> So i am confused what's this for

[asciicast](https://asciinema.org/a/TTnzdjM8ctCi7mfyQ65zazmky)

The `test` queue spec is 2CPU, 1Gi Memory.
Job spec is 1CPU, 500Mi Memory with 5replicas.
When I submit the Job, all tasks will be scheduled!!! So Job is using 5CPU, 2500Mi Memory in `test` queue!

**Comment by zen-xu** [comment]
> > So i am confused what's this for
> 
> [asciicast](https://asciinema.org/a/TTnzdjM8ctCi7mfyQ65zazmky)
> 
> The `test` queue spec is 2CPU, 1Gi Memory.
> Job spec is 1CPU, 500Mi Memory with 5replicas.
> When I submit the Job, all tasks will be scheduled!!! So Job is using 5CPU, 2500Mi Memory in `test` queue!

I hope 2 tasks will be running and others will be pending

**Comment by hzxuzhonghu** [comment]
That's right, i remember we have fixed this issue. So that's why i am confused. Let me have a look

**Comment by hzxuzhonghu** [comment]
@zen-xu I suspect you do not enable `enqueue ` action right? 

This pr looks good, bu expect file a fix into master branch first, and cherry-pick back to release-1.0 and release-0.4 after that.

**Comment by zen-xu** [comment]
> @zen-xu I suspect you do not enable `enqueue ` action right? 
> 
> 
> 
> This pr looks good, bu expect file a fix into master branch first, and cherry-pick back to release-1.0 and release-0.4 after that.

enqueue is enabled

**Comment by hzxuzhonghu** [comment]
IC, the minavailable is 1

**Comment by zen-xu** [comment]
> IC, the minavailable is 1

Yes，I forget to mention it🤦🏻‍♂️

File: pkg/scheduler/plugins/proportion/proportion.go [hunk 1]
```diff
@@ -161,11 +166,15 @@ func (pp *proportionPlugin) OnSessionOpen(ssn *framework.Session) {
 			oldDeserved := attr.deserved.Clone()
 			attr.deserved.Add(remaining.Clone().Multi(float64(attr.weight) / float64(totalWeight)))
 
-			if attr.request.Less(attr.deserved) {
+			if attr.capability != nil && !attr.deserved.LessEqualStrict(attr.capability) {
+				attr.deserved = helpers.Min(attr.deserved, attr.capability)
+				attr.deserved = helpers.Min(attr.deserved, attr.request)
+				meet[attr.queueID] = struct{}{}
+				klog.V(4).Infof("queue <%s> is meet cause of the capability", attr.name)
+			} else if attr.request.Less(attr.deserved) {
```
Comment (hzxuzhonghu) [inline pkg/scheduler/plugins/proportion/proportion.go:174]: lgtm

maybe we can clean it a little bit more, remove the less check

```
if attr.capability != nil{
    attr.deserved = helpers.Min(attr.deserved, attr.capability)
}

attr.deserved = helpers.Min(attr.deserved, attr.request)
```

**Pull request merged**